### PR TITLE
Update mcr.cpp

### PR DESCRIPTION
**Pull request by Hedwig-w13**

### CODE REVIEW

File: mcr.cpp [hunk 1]
```diff
@@ -1,57 +1,67 @@
...
+        
+        cout << "Which cell to mark? i:[1..3], j:[1..3]: ";
+        cin >> i >> j;
+        
+        i--;         j--;
+        
+        if (game[i][j] != ' ') { 
+            cout << "please try another cell" << endl;
+            n--; 
+            continue;
```
Comment (zehaosu0301) [inline mcr.cpp:41]: When I repeatedly play in a place, he will prompt me to please try another cell and skip the current turn, which will result in a party being able to play twice in a row.

File: mcr.cpp [hunk 2]
```diff
@@ -1,57 +1,67 @@
...
-else 
- game[i][j] = 'O'; 
- if (isWin(game)){
- cout << "Win!" << endl;
- break; // need to terminate the problem
- } 
- } 
-if (i==3) // all celles with i=0 to 2 have been inputted above but no 
-winner yet
- cout << "Tie!" << endl;
```
Comment (scottyyyyyy) [inline mcr.cpp:51]: The winner should be marked, thus i think the comment should tell who is the winner.

File: mcr.cpp [hunk 3]
```diff
@@ -1,57 +1,67 @@
...
-Player 1 first.
-cout << "X = Player 1" << endl << "O = Player 2" << endl;
- for (int n=0; n<9; n++){
- turn = !turn; // use the not-operator to change true to false or 
-false to true.
- if (turn == false) 
- cout << "Player 1: "; 
- else
- cout << "Player 2: "; 
- cout << "Which cell to mark? i:[1..3], j:[1..3]: ";
```
Comment (scottyyyyyy) [inline mcr.cpp:38]: When the gamers input the decision location,firstly we should do a judge whether the location was be occupied.This is a bascially rule.

**Review by scottyyyyyy — CHANGES_REQUESTED** [review]

File: mcr.cpp [hunk 4]
```diff
@@ -1,57 +1,67 @@
...
-== 'X' || game[0][0] == 'O')) win = true; 
-if (game[0][1] == game[1][1] && game[1][1] == game[2][1] && (game[0][1] 
-== 'X' || game[0][1] == 'O')) win = true; 
-if (game[0][2] == game[1][2] && game[1][2] == game[2][2] && (game[0][2] 
-== 'X' || game[0][2] == 'O')) win = true; 
-// diagonal
-if (game[0][0] == game[1][1] && game[1][1] == game[2][2] && (game[0][0] 
-== 'X' || game[0][0] == 'O')) win = true; 
-if (game[0][2] == game[1][1] && game[1][1] == game[2][0] && (game[0][2] 
-== 'X' || game[0][2] == 'O')) win = true;
```
Comment (smzywdsc) [inline mcr.cpp:20]: Ok

**Review by zehaosu0301 — APPROVED** [review]

File: mcr.cpp (hunk 3 above)
Comment (morannn) [inline mcr.cpp:38]: there use arrays 1to3 But the following uses 0to2

File: mcr.cpp [hunk 5]
```diff
@@ -1,57 +1,67 @@
...
-cout << "X = Player 1" << endl << "O = Player 2" << endl;
- for (int n=0; n<9; n++){
- turn = !turn; // use the not-operator to change true to false or 
-false to true.
- if (turn == false) 
- cout << "Player 1: "; 
- else
- cout << "Player 2: "; 
- cout << "Which cell to mark? i:[1..3], j:[1..3]: "; 
- cin >> i >> j;
```
Comment (Lily20001203) [inline mcr.cpp:39]: I think that we need to add the statement to review the input which is valid or not

**Review by morannn — APPROVED** [review]

**Pull request merged**